### PR TITLE
clazy, 32bit build fix

### PR DESCRIPTION
**Pull request by Begasus**
Add upstream patches (prepares for building with LLVM18)

### CODE REVIEW

**Comment by Begasus** [comment]
Added LLVM18 (upstream) patches, these don't collide when building with LLVM17, all tests passed on checking 64bit with both clang versions.

File: dev-util/clazy/patches/clazy-1.11.patchset [hunk 1]
```diff
@@ -285,37 +285,179 @@ index b0da926..1ba1126 100644
...
+   add_executable(clazy-standalone ${CLAZY_STANDALONE_SRCS})
+ 
+-- 
+2.45.1
+
+
+From 6fd688c72eda9a1ccf76ac697cc8735ebb7b70fd Mon Sep 17 00:00:00 2001
+From: Schrijvers Luc <begasus@gmail.com>
+Date: Thu, 13 Jun 2024 13:50:31 +0200
+Subject: Disable PCH for 32bit
```
Comment (Begasus) [inline dev-util/clazy/patches/clazy-1.11.patchset:441]: @korli this isn't stricktly needed for 64bit, if so I could move this patch out and only apply it on 32bit only.

Comment (Begasus) [reply dev-util/clazy/patches/clazy-1.11.patchset:441]: Could be done later if need be. :)

**Comment by Begasus** [comment]
Checked build on 32bit (R1B4), KDevelop picks it up after installation and with path to the documention set it detects 94 checkes (same as on 64bit (R1B4).

**Comment by korli** [comment]
Thanks!

**Pull request merged**